models/concerns/as_json.rb

```diff
@@ -29,6 +29,17 @@ def empty_json(options = nil)
   end
 end
 
+module ServiceAsJson
+  extend ActiveSupport::Concern
+
+  def as_json(options = nil)
+    service = super
+
+    service.delete('activities') if activities.empty?
+    service
+  end
+end
+
```

Comment:
Why don't we do this systematically for all dumped objects? Do we ever want to dump empty arrays? If we do it `ServiceAsJson` way, we go back to square-zero in the maintainability and automation sense.

Reply:
Dumping an empty array is not really an issue in the general case. Empty arrays are generated through the has_many active_hash method.

The special cases are for service activities and vehicle sequence_timewindows because these fields are mutually exclusive with their singular version...

Reply:
See the comments both in models/service and models/vehicle. These classes should be remove as soon as we merge singular and plural fields